libraries/UHS_host/UHS_ADK/UHS_ADK_INLINE.h

```diff
@@ -0,0 +1,222 @@
+/* Copyright (C) 2011 Circuits At Home, LTD. All rights reserved.
+
+This software may be distributed and modified under the terms of the GNU
+General Public License version 2 (GPL2) as published by the Free Software
+Foundation and appearing in the file GPL2.TXT included in the packaging of
+this file. Please note that GPL2 Section 2[b] requires that all works based
+on this software must also be made publicly available under the terms of
+the GPL2 ("Copyleft").
+
+Contact information
+-------------------
+
+Circuits At Home, LTD
+Web      :  http://www.circuitsathome.com
+e-mail   :  support@circuitsathome.com
+ */
+
+/* Google ADK interface */
+
+#if defined(LOAD_UHS_ADK) && defined(__UHS_ADK_H__) && !defined(UHS_ADK_LOADED)
+#define UHS_CDC_ACM_LOADED
+
+ADK::ADK(UHS_USB_HOST_BASE *p, const char* manufacturer,
+        const char* model,
+        const char* description,
+        const char* version,
+        const char* uri,
+        const char* serial) :
+
+/* ADK ID Strings */
+manufacturer(manufacturer),
+model(model),
+description(description),
+version(version),
+uri(uri),
+serial(serial),
+ready(false) {
```

Comment:
Why not just have the end user #define the strings, etc. (unless you want to have > 1 host on ADK) This will make the code more compact. Just make sure you prepend them properly to avoid namespace collisions. For example UHS_ADK_MANUFACTURER, etc. Provide defaults for these as well. Otherwise send a structure if you want to support multiple instances.